src/tests/test_ucc_verifier.cpp

```diff
@@ -0,0 +1,76 @@
+#include <gtest/gtest.h>
+
+#include "algorithms/algo_factory.h"
+#include "algorithms/ucc/ucc_verifier/ucc_verifier.h"
+#include "config/indices/type.h"
+#include "datasets.h"
+
+namespace tests {
+namespace onam = config::names;
+
+struct UCCVerifyingParams {
+private:
+    algos::StdParamsMap params;
+    size_t num_clusters_not_ucc_ = 0;
+    size_t num_rows_not_ucc_ = 0;
+    std::vector<model::PLI::Cluster> clusters_not_ucc_;
+
+public:
+    UCCVerifyingParams(config::IndicesType column_indices, size_t const num_clusters_not_ucc,
+                       size_t const num_rows_not_ucc,
+                       std::vector<model::PLI::Cluster> clusters_not_ucc, char const* dataset,
+                       char const separator = ',', bool const has_header = true)
+        : params({{onam::kIndices, std::move(column_indices)},
+                  {onam::kCsvPath, test_data_dir / dataset},
+                  {onam::kSeparator, separator},
+                  {onam::kHasHeader, has_header},
+                  {onam::kEqualNulls, true}}),
+          num_clusters_not_ucc_(num_clusters_not_ucc),
+          num_rows_not_ucc_(num_rows_not_ucc),
+          clusters_not_ucc_(std::move(clusters_not_ucc)) {}
+
+    const algos::StdParamsMap& GetParams() const {
+        return params;
+    }
+
+    size_t GetExpectedNumClustersNotUCC() const {
+        return num_clusters_not_ucc_;
+    }
+
+    size_t GetExpectedNumRowsNotUCC() const {
+        return num_rows_not_ucc_;
+    }
+
+    const std::vector<model::PLI::Cluster>& GetExpectedClustersNotUCC() const {
+        return clusters_not_ucc_;
+    }
+};
+
+class TestUCCVerifying : public ::testing::TestWithParam<UCCVerifyingParams> {};
+
+TEST_P(TestUCCVerifying, DefaultTest) {
+    auto const& p(GetParam());
+    auto mp = algos::StdParamsMap(p.GetParams());
```

Comment:
Strange cast, `p.GetParams()` already has `algos::StdParamsMap` type, there is no need to cast it. Moreover, this `mp` variable is not needed at all, since it is only used in the line below, `p.GetParams()` could be passed there right away

Reply:
We need `mp` variable because `CreateAndLoadAlgorithm` accepts a rvalue-reference and `p.GetParams` returns const l-value reference